### PR TITLE
`io.grpc` version to 1.39, `io.netty` version to 4.1.66.Final

### LINKED ISSUE #1560: Upgrade netty
**Summary**
Please upgrade netty to the latest version. Should also ensure BouncyCastle is present and updated so that TLSv1.3 gets [enabled](https://github.com/netty/netty/pull/11193).

**Possible resolution**
Upgrade netty

### PR DESCRIPTION
**Pull request by tinker-michaelj**
**Description**:
- Update _pom.xml_ properties: 
    1. `grpc.version -->> 1.39`
    2. `hapi.proto -->> 0.17.0-SNAPSHOT`
    2. `netty.version -->> 4.1.65.Final`
- Add explicit dependency management for the `netty-codec-http2` and `netty-handler-proxy` artifacts.
 
**Related issue(s)**:
- Closes #1560

### CODE REVIEW

**Review by qnswirlds — DISMISSED** [review]

**Review by steven-sheehy — CHANGES_REQUESTED** [review]
Have you reviewed the list of [netty config](https://github.com/hashgraph/hedera-services/blob/020bdfe1300a83a42391c265e6fc153eeec163b5/hedera-node/src/main/java/com/hedera/services/grpc/ConfigDrivenNettyFactory.java) being used to see if it needs adjustment? Specifically there was a large change in netty around flow control. You probably don't need to set the flow control window anymore because if you do it overrides their improved logic (please confirm!). Also, please test the list of supported ciphers as they should either be removed or widened to support newer ones. The TLS_DHE_RSA_WITH_AES_256_GCM_SHA384 is deprecated as insecure in most places and should probably be removed.

File: pom.xml [hunk 1]
```diff
@@ -160,13 +160,13 @@
...
+    <grpc.version>1.39.0</grpc.version>
     <jackson-databind.version>2.9.10.7</jackson-databind.version>
     <javax.annotation-api.version>1.3.2</javax.annotation-api.version>
-    <hapi-proto.version>0.16.0-rc.1</hapi-proto.version>
+    <hapi-proto.version>0.17.0-SNAPSHOT</hapi-proto.version>
     <javax-inject.version>1</javax-inject.version>
     <log4j.version>2.13.2</log4j.version>
-    <netty.version>4.1.51.Final</netty.version>
+    <netty.version>4.1.65.Final</netty.version>
     <swirlds.version>0.16.0</swirlds.version>
```
Comment (steven-sheehy) [inline pom.xml:170]: Is there any reason not to use 4.1.66.Final?

Comment (qnswirlds) [reply pom.xml:170]: netty `4.1.66.Final` is fine with me.

**Comment by steven-sheehy** [comment]
The change to flow control window was actually in [grpc-java 1.30](https://github.com/grpc/grpc-java/releases/tag/v1.30.0). You can probably keep the property but set it to 0 by default so it uses the new Bandwidth Delay Product (BDP) by default. Then you can conditionally set it if it's non-zero to re-enable the old behavior based upon testing in a live environment.

File: pom.xml (hunk 1 above)
Comment (tinker-michaelj) [reply pom.xml:170]: Changed to `4.1.66.Final`.

**Comment by tinker-michaelj** [comment]
> The change to flow control window was actually in [grpc-java 1.30](https://github.com/grpc/grpc-java/releases/tag/v1.30.0), not Netty. So this was missed previously. You can probably keep the property but set it to 0 by default so it uses the new Bandwidth Delay Product (BDP) by default. Then you can conditionally set it if it's non-zero to re-enable the old behavior based upon testing in a live environment.

Discussed with priority-setters and created https://github.com/hashgraph/hedera-services/issues/1898 to follow up on this; although it seems very likely we can get some benefit, validating changes needs cycles we don't currently have.

**Review by steven-sheehy — APPROVED** [review]
Sounds good

**Review by qnswirlds — APPROVED** [review]

**Pull request merged**